tests/functional/aws-node-sdk/test/bucket/testBucketStress.js

```diff
@@ -0,0 +1,55 @@
+import { S3 } from 'aws-sdk';
+import { times, timesSeries, waterfall } from 'async';
+
+import getConfig from '../support/config';
+
+const bucket = `stress-test-bucket-${Date.now()}`;
+const text = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
+const objectCount = 100;
+const loopCount = 10;
+
+function putObjects(s3, cb) {
+    let count = 0;
+    times(objectCount, (i, next) => {
+        const params = { Bucket: bucket, Key: `foo${i}`, Body: text };
+        s3.putObject(params, error => {
+            if (error || ++count === objectCount) {
+                cb(error);
```

Comment:
If the count is off but error is null, I don't think we will see anything here.  The cb will just be called with null so the error will be hidden (I suppose until we try to delete it but it will be hard to figure out what is going on at that point).  I think you should create the error object to make sure it exists if the count is off and log a message as well (just by using process.stdout.write).